src/iuse.cpp

```diff
@@ -6439,20 +6539,788 @@ struct npc_photo_def : public JsonDeserializer, public JsonSerializer {
+        found_item_aim_point = !item.is_null();
+    }
+
+    const ter_id ter_aim = g->m.ter( aim_point );
+    const furn_id furn_aim = g->m.furn( aim_point );
+
+    if( !description_figures_status.empty() ) {
+        std::string names = enumerate_as_string( description_figures_status.begin(),
+                            description_figures_status.end(),
+        []( const std::pair<std::string, std::string> &it ) {
+            return "<color_light_blue>" + it.first + "</color>";
+        } );
+
+        photo.name = names;
+        photo_text += names + ".";
+
+        for( const auto &figure_status : description_figures_status ) {
+            photo_text += "\n\n<color_light_blue>" + figure_status.first + "</color> " + figure_status.second;
+        }
+    } else if( found_vehicle_aim_point ) {
+        const optional_vpart_position veh_part_pos = g->m.veh_at( aim_point );
+        const std::string veh_name = "<color_light_blue>" + veh_part_pos->vehicle().disp_name() +
+                                     "</color>";
+        photo.name = veh_name;
+        photo_text += veh_name + ".";
+        map_deincrement_or_erase( obj_coll.vehicles, veh_name );
+    } else if( found_furniture_aim_point || found_item_aim_point )  {
+        std::string item_name = colorized_item_name( item );
+        if( found_furniture_aim_point ) {
+            furn_desc = trap_name + furn_desc + field_desc;
+            photo.name = furn_desc;
+            photo_text += photo.name + ".";
+            map_deincrement_or_erase( obj_coll.furniture, furn_desc );
+        } else if( found_item_aim_point ) {
+            item_name = trap_name + item_name + field_desc;
+            photo.name = item_name;
+            photo_text += item_name + ". " + string_format( _( "It lies on the %s." ),
+                          ter_name );
+            map_deincrement_or_erase( obj_coll.items, item_name );
+        }
+        if( found_furniture_aim_point && !furn_aim->description.empty() ) {
+            photo_text += "\n\n<color_yellow>" + furn_aim->name() + "</color>:\n" + furn_aim->description;
+        }
+        if( found_item_aim_point ) {
+            photo_text += "\n\n" + item_name + ":\n" + colorized_item_description( item );
+        }
+    } else {
+        ter_name = trap_name + ter_name + field_desc;
+        photo.name = ter_name;
+        photo_text += photo.name + ".";
+        map_deincrement_or_erase( obj_coll.terrain, ter_name );
+        map_deincrement_or_erase( obj_coll.furniture, ter_name );
+
+        if( !ter_aim->description.empty() ) {
+            photo_text += "\n\n" + photo.name + ":\n" + ter_aim->description;
+        }
+    }
+
+    if( !obj_coll.items.empty() ) {
+        std::string obj_list = enumerate_as_string( obj_coll.items.begin(), obj_coll.items.end(),
+                               format_object_pair );
+        photo_text += "\n\n" + string_format( _( "There is something lying on the ground: %s." ),
```

Comment:
This would be even better:
```suggestion
        photo_text += "\n\n" + string_format( _( "There is something lying on the %s: %s." ), ter_name
```

Reply:
This one is final line: `There is something lying on the ground: flashlight, apple and crowbar`, so every item from different tile. I decided not include terrain name for every item, unless there is notable terrain under them.

Reply:
Yeah that makes sense